src/writers/lsbi.c

```diff
@@ -0,0 +1,199 @@
+        }
+
+        free(pattern_data);
+        return 0;
+    }
+
+    size *= 8;
+
+    size_t bits_needed = size;
+    size_t bytes_needed = (3 * bits_needed + 1) / 2;
+
+    uint8_t *data = malloc(bytes_needed);
+    if (data == NULL)
+    {
+        perror("Memory allocation failed");
+        free(pattern_data);
+        return 0;
+    }
+    
+    if (fread(data, sizeof(uint8_t), bytes_needed, output) != bytes_needed)
+    {
+        if (feof(output))
+        {
+            fprintf(stderr, "Error: Output file is too small to hold the data.\n");
+        }
+        else
+        {
+            perror("Error reading output file");
+        }
+        free(pattern_data);
+        free(data);
+        return 0;
+    }
+    if (fseek(output, start_offset, SEEK_SET) != 0)
+    {
+        perror("fseek failed");
+        free(pattern_data);
+        free(data);
+        return 0;
+    }
+    
+    size_t curr_byte = 0;
+    size_t curr_byte_input = 0;
+    size_t total_bits = bits_needed;
+    while(curr_byte_input < total_bits)
+    {
+        uint8_t byte = input[curr_byte_input / 8];
+
+        //LOG("Embedding byte: %02X\n", byte);
+        for (int i = 7 ; i >= 0 && curr_byte_input < total_bits ; i--)
+        {
+            uint8_t bit = (byte >> i) & 1;
+            //LOG("Embedding bit: %d\n", bit);
+
+            if((curr_byte % 3) == 1) // it will be a R
+            {
+                curr_byte++;
+                if (curr_byte >= bytes_needed)
+                {
+                    fprintf(stderr, "Error: Overflow while skipping R\n");
+                    free(pattern_data);
+                    free(data);
+                    return 0;
+                }
+            }
+
+            if (curr_byte >= bytes_needed)
+            {
+                fprintf(stderr, "Error: Overflow before accessing data\n");
+                free(pattern_data);
+                free(data);
+                return 0;
+           }
+
+            uint8_t value = data[curr_byte];
+            uint8_t last3 = LSB(value, 3);
+            int pattern_index = last3 >> 1;
+            //LOG("Pattern of bit: %d\n", pattern_index);
+            (bit == (last3 & 1)) ? pattern[pattern_index]-- : pattern[pattern_index]++;
+            
+            data[curr_byte] = (data[curr_byte] & 0xFE) | bit;
+            curr_byte++;
+            curr_byte_input++;
+            if (curr_byte >= bytes_needed)
+            {
+                fprintf(stderr, "Error: Overflow during embedding\n");
+                free(pattern_data);
+                free(data);
+                return 0;
+            }
+        }
+    }
+    int has_to_invert = 0;
+    for(int i = 0; i < PATTERN_SIZE; i++)
+    {
+        if(pattern[i] > 0)
+        {
+            has_to_invert++;
+            pattern[i] = 1;
+        } else {
+            pattern[i] = 0;
+        }
+    }
+
+    LOG("Number of patterns to invert: %d\n", has_to_invert);
+    
+    if(has_to_invert > 0)
```

Comment:
Y acá solo es el bool

```suggestion
    LOG("Invertion needed: %s\n", has_to_invert ? "true" : "false");

    if(has_to_invert)
```